src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -448,6 +474,18 @@ void Device::execute(const std::string &name,
     cmd_buf.submit_and_wait();
 }
 
+std::optional<std::uint32_t> Device::find_queue_family_index_if(
+    const std::function<bool(const std::uint32_t index, const VkQueueFamilyProperties &)> &criteria_lambda) {
+    std::uint32_t index = 0;
+    for (const auto queue_family : vk_tools::get_all_physical_device_queue_family_properties(m_physical_device)) {
+        if (queue_family.queueCount > 0 && criteria_lambda(index, queue_family)) {
```

Comment:
```suggestion
        if (criteria_lambda(index, queue_family)) {
```
A queue family is guaranteed to have at least one queue.

Reply:
Done.